File: CareConnect/CareConnect.Models/Requests/ClientInsertRequest.cs
using System;
using System.Text.Json.Serialization;

namespace CareConnect.Models.Requests
{
    public class ClientInsertRequest
    {
        public bool EmploymentStatus { get; set; }

        [JsonIgnore]
        public DateTime ModifiedDate { get; set; } = DateTime.Now;

        public UserInsertRequest User { get; set; }
    }
}

File: UI/careconnect_admin/lib/models/requests/client_insert_request.dart
import 'package:careconnect_admin/models/requests/user_insert_request.dart';

import 'package:json_annotation/json_annotation.dart';

part 'client_insert_request.g.dart'; // Don't open or edit this

@JsonSerializable()
class ClientInsertRequest {
  final bool employmentStatus;
  final UserInsertRequest user;

  ClientInsertRequest({required this.employmentStatus, required this.user});

  factory ClientInsertRequest.fromJson(Map<String, dynamic> json) =>
      _$ClientInsertRequestFromJson(json);

  Map<String, dynamic> toJson() => _$ClientInsertRequestToJson(this);
}
